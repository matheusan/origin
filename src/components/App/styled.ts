import styled, { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;

    background-color: ${props => props.theme.colors.blueGray10}
  }
`;

const Container = styled.div`
  width: 100%;
`;

export { GlobalStyle, Container };
