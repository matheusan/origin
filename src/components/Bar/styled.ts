import styled from 'styled-components';

export const Bar: JSX.Element = styled.div`
  display: flex;
  top: 0;
  left: 0;
  right: 0;
  height: 80px;

  padding: 24px 56px;
  background-color: ${props => props.theme.colors.neutralWhite};
  align-items: center;

  @media only screen and (max-width: 380px) {
    height: 56px;
  }
`;
