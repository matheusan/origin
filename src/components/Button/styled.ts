import styled from 'styled-components';

export const Button: JSX.Element = styled.button`
  display: block;
  width: ${props => (props.secondary || props.flex ? '100%' : '320px')};
  height: ${props => (props.secondary ? '40px' : '56px')};
  margin: 0 auto;
  border-radius: ${props => (props.secondary ? '8px' : '32px')};
  border: 0;
  outline: 0;
  cursor: pointer;

  background-color: ${props =>
    props.secondary
      ? props.theme.colors.blueGray10
      : props.theme.colors.brandColorPrimary};

  color: ${props =>
    props.secondary
      ? props.theme.colors.brandColorSecondary
      : props.theme.colors.neutralWhite};

  font-family: ${props => props.theme.fonts.primary};
  font-size: 16px;
  line-height: ${props => (props.secondary ? '40px' : '54px')};
  text-align: center;

  &:hover {
    background-color: ${props =>
      props.secondary
        ? props.theme.colors.neutralWhite
        : props.theme.colors.brandColorSecondary};
  }
`;
