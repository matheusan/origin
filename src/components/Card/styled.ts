import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: stretch;
  align-content: stretch;

  margin-top: 24px;
  margin-bottom: 32px;

  border-radius: 8px;
  border: 1px solid ${props => props.theme.colors.blueGray50};
`;

export const Single: JSX.Element = styled.div`
  padding: 16px 28px;

  @media only screen and (max-width: 360px) {
    padding: 12px 32px;
  }
`;

export const Top: JSX.Element = styled.div`
  padding: 24px 32px;

  @media only screen and (max-width: 360px) {
    padding: 24px 27px;
  }
`;
export const Bottom: JSX.Element = styled.div`
  padding: 24px 32px;

  background-color: ${props => props.theme.colors.blueGray10};

  line-height: 16px;
  font-weight: 400;
  font-size: 12px;
  font-family: ${props => props.theme.fonts.primary};
  color: ${props => props.theme.colors.blueGray900};
`;
