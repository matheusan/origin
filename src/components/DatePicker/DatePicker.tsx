import * as React from 'react';
import { useEffect, useCallback } from 'react';
import { format } from 'date-fns';

import * as S from './styled';
import Icon from '../Icon';

export interface DatePickerProps {
  date?: Date;
  onClickNext?: () => null;
  onClickPrev?: () => null;
}

const DatePicker: React.SFC<DatePickerProps> = ({
  date = new Date(),
  onClickNext,
  onClickPrev
}) => {
  const handleKeyPress = useCallback(
    event => {
      const { keyCode } = event;
      if (keyCode === 39) {
        onClickNext();
      }
      if (keyCode === 37) {
        onClickPrev();
      }
    },
    [onClickNext, onClickPrev]
  );

  useEffect(() => {
    window.addEventListener('keydown', handleKeyPress);

    return () => {
      window.removeEventListener('keydown', handleKeyPress);
    };
  }, [handleKeyPress]);

  return (
    <S.Wrapper>
      <S.Button onClick={onClickPrev}>
        <Icon name="arrow" />
      </S.Button>
      <S.MonthYear>
        <strong>{format(date, 'MMMM')}</strong>
        {format(date, 'yyyy')}
      </S.MonthYear>
      <S.Button direction="right" onClick={onClickNext}>
        <Icon name="arrow" />
      </S.Button>
    </S.Wrapper>
  );
};

export default DatePicker;
