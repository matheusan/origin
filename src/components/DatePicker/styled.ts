import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: stretch;
  align-content: stretch;

  max-height: 56px;

  border: 1px solid ${props => props.theme.colors.blueGray50};
  background: ${props => props.theme.colors.neutralWhite};
  padding: 3px 5px;
`;

export const Button: JSX.Element = styled.button`
  margin: 0;
  padding: 3px 10px;
  outline: none;
  background-color: transparent;
  border: 0;
  cursor: pointer;

  ${props =>
    props.direction && props.direction === 'right'
      ? `> img { transform: rotate(180deg); }`
      : ''}
`;

export const MonthYear: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;

  margin: 0 10px;
  line-height: 24px;
  font-size: 16px;
  font-family: ${props => props.theme.fonts.primary};
  color: ${props => props.theme.colors.blueGray900};
`;
