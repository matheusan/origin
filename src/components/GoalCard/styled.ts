import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: stretch;
  align-content: stretch;

  width: 260px;
  height: 250px;

  padding: 24px;
  margin-bottom: 2em;

  background: ${props => props.theme.colors.neutralWhite};

  border-radius: 8px;
  box-shadow: 0px 16px 32px rgba(30, 42, 50, 0.38);

  @media only screen and (max-width: 380px) {
    margin: 0 auto;
    border-radius: 4px;
    padding: 24px;
  }
`;
