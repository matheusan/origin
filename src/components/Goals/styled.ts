import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  max-width: 1300px;
  margin: 0 auto;
`;
