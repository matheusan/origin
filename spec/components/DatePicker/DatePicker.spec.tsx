import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import DatePicker from '../../../src/components/DatePicker';
import { Button } from '../../../src/components/DatePicker/styled';
import { originDefaultTheme } from '../../../src/theme/originDefaultTheme';
import { mountWithTheme } from '../../testutils';

describe('DatePicker', () => {
  const map = {};
  const removeEventListener = jest.fn();
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    global.removeEventListener = removeEventListener;
    global.addEventListener = jest.fn((event, cb) => {
      map[event] = cb;
    });
  });

  beforeEach(() => {
    const mockDate = new Date(1604977200000);
    jest.spyOn(Date, 'now').mockReturnValue(+mockDate);
    wrapper = shallow(<DatePicker />);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire up function when click next', () => {
    const up = jest.fn();
    wrapper = shallow(<DatePicker onClickNext={up} />);
    const nextButton = wrapper.find(Button).at(1);
    nextButton.simulate('click');
    expect(up).toHaveBeenCalled();
  });

  it('should fire down function when click prev', () => {
    const down = jest.fn();
    wrapper = shallow(<DatePicker onClickPrev={down} />);
    const nextButton = wrapper.find(Button).at(0);
    nextButton.simulate('click');
    expect(down).toHaveBeenCalled();
  });

  describe('when using the keyboard', () => {
    it('should fire up function when arrow right', () => {
      const up = jest.fn();
      mountWithTheme(<DatePicker onClickNext={up} />, originDefaultTheme);
      map.keydown({ keyCode: 39 });
      expect(up).toHaveBeenCalled();
    });

    it('should fire down function when arrow left', () => {
      const down = jest.fn();
      mountWithTheme(<DatePicker onClickPrev={down} />, originDefaultTheme);
      map.keydown({ keyCode: 37 });
      expect(down).toHaveBeenCalled();
    });
  });

  it('should remove eventlistener when unmounting', () => {
    const mounted = mountWithTheme(<DatePicker />, originDefaultTheme);
    expect(map.keydown).toBeDefined();

    mounted.unmount();
    expect(removeEventListener).toHaveBeenCalled();
  });
});
