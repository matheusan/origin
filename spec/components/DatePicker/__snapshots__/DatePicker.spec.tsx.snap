// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders 1`] = `
<styled.div>
  <styled.button>
    <Icon
      name="arrow"
    />
  </styled.button>
  <styled.div>
    <strong>
      November
    </strong>
    2020
  </styled.div>
  <styled.button
    direction="right"
  >
    <Icon
      name="arrow"
    />
  </styled.button>
</styled.div>
`;
